src/main/java/org/mule/extension/file/internal/DirectoryListener.java

```diff
@@ -277,7 +277,10 @@ private void listen(SourceCallback<InputStream, ListenerFileAttributes> sourceCa
         }
       }
     } catch (Exception e) {
-      sourceCallback.onSourceException(e);
+      LOGGER.error(format("Directory listener at flow '%s' found exception listening for File System changes. "
```

Comment:
Are we sure that the exception thrown may not be because the directory has been deleted?. If that may be the case it seems the source will just stop working and never try to reconnect